src/librbd/io/ImageRequestWQ.cc

```diff
@@ -525,6 +537,23 @@ void ImageRequestWQ<I>::set_require_lock(Direction direction, bool enabled) {
 }
 
 template <typename I>
+void ImageRequestWQ<I>::apply_qos_iops_limit(uint64_t limit) {
+  iops_throttle->set_max(limit);
+  iops_throttle->set_average(limit);
+}
+
+template <typename I>
+void ImageRequestWQ<I>::handle(int r, ImageRequest<I> *item) {
```

Comment:
Should be private and named appropriately to be informative

Reply:
ok, what about ```iops_throttle_callback```

Reply:
I'd prefer to allow the user to not require a fixed public callback -- w/ templating, there shouldn't be a need.

Reply:
it's already private now, sorry, but I am not sure what's your suggestion here.